keanu-project/src/main/java/io/improbable/keanu/tensor/NumberTensor.java

```diff
@@ -29,6 +29,39 @@
     
     T abs();
 
+    /**
+     * Find the index into the flattened array of the tensor of the largest value, e.g.
+     * <pre>
+     * DoubleTensor tensor = DoubleTensor.arange(0, 6).reshape(2, 3);
+     * // [[0., 1., 2.],
+     * //  [3., 4., 5.]]
+     * IntegerTensor max = tensor.argMax();
+     * // [[5]]
+     * </pre>
+     *
+     * @return A scalar tensor with the index of the largest value
+     */
+    IntegerTensor argMax();
+
+    /**
+     * Find the indices into the tensor of the largest values in a specified axis (dimension), e.g.
+     * <pre>
+     * DoubleTensor tensor = DoubleTensor.arange(0, 6).reshape(2, 3);
+     * // [[0., 1., 2.],
+     * //  [3., 4., 5.]]
+     * IntegerTensor maxesFor0 = tensor.argMax(0);
+     * // [[1, 1, 1]]
+     * IntegerTensor maxFor1 = tensor.argMax(1);
+     * // [[2, 2]]
+     * </pre>
+     *
+     * @param axis The axis (dimension) to find the largest values in
+     * @return A tensor where each value is the location of the maximum value in the vector at that location in the
+     *         specified dimension in the original tensor
+     * @see <a href="https://www.geeksforgeeks.org/numpy-argmax-python/">An article about argmax over an axis in numpy</a>
+     */
+    IntegerTensor argMax(int axis);
```

Comment:
Is this returning a rank 1 vector? I think you'll need to change this to a rank 2

Reply:
Yep - the code as it stands will get confused by anything rank < 2

Reply:
Wasn't sure exactly what restrictions were here, I'll fix that up!